fastly/basictypes_helper.go

```diff
@@ -1,27 +1,7 @@
 package fastly
 
-// String is a helper that returns a pointer to the string value passed in.
-func String(v string) *string {
-	return &v
-}
-
-// Int is a helper that returns a pointer to the int value passed in.
-func Int(v int) *int {
-	return &v
-}
-
-// Uint is a helper that returns a pointer to the uint value passed in.
-func Uint(v uint) *uint {
-	return &v
-}
-
-// Uint8 is a helper that returns a pointer to the uint8 value passed in.
-func Uint8(v uint8) *uint8 {
-	return &v
-}
-
-// Bool is a helper that returns a pointer to the bool value passed in.
-func Bool(v bool) *bool {
+// ToPointer converts T to *T.
+func ToPointer[T string | int | uint | uint8 | bool | CacheSettingAction | Compatibool | ERLAction | ERLLogger | ERLWindowSize | RequestSettingAction | RequestSettingXFF | S3AccessControlList | S3Redundancy | S3ServerSideEncryption](v T) *T {
```

Comment:
Curious why you specify all these types rather than using `any`?

I wonder also if you could use the `~` sigil on the primitive types to get the underlying types?  If not `any` then `~string | ~int | ~uint | ~uint8 | ~bool`?

Reply:
Haha never mind, I saw the follow-up PR.